src/renderers/webgl/WebGLProgram.js

```diff
@@ -173,9 +173,13 @@ function filterEmptyLine( string ) {
 
 function replaceLightNums( string, parameters ) {
 
+	var numSpotLightCoords = parameters.shadowMapEnabled ? parameters.numSpotLights : parameters.numSpotLightMaps;
```

Comment:
What is the purpose of this line?

Reply:
That value will be defined as `NUM_SPOT_LIGHT_COORDS ` in glsl, to know the size of spot light coords in the shaders (which are used by both texture mapping and shadow mapping)